### DISCUSSION TITLE
Wheels traction

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#97] [Q&A] [answered]
Is there a way to find the current traction of a wheel (wheeled vehicle) with the ground? Something like 0 being no traction at all to 1 being full traction or something. This data could help with adding tire trails and dust effects to vehicles.

Another option would be to have a dot product of the forward direction of the vehicle with its linear velocity. Just wonder if there is a way to find the slip per wheel.

**Comment by jrouwe** [comment, marked as answer]
WheelWV (which inherits from Wheel) has mLongitudinalSlip and mLateralSlip which can be used for this.

**Reply by LeXXik** [reply]
Hmm, sorry, but how do I access `WheelWV` from `VehicleConstraint`? There is `.GetWheel(index)`, but it returns `Wheel`, which doesn't have those properties.

I am creating wheels this way:
```js
const wheel = new Jolt.WheelSettingsWV();
vehicleConstraintSettings.mWheels.push_back(wheel);
```

**Reply by jrouwe** [reply]
You can cast Wheel to WheelWV by doing:

```
const wheel = vehicleConstraint.GetWheel(xxx);
const wheelwv = Jolt.castObject(wheel, Jolt.WheelWV);
```

(so not on the vehicleConstraintSettings but on the created constraint)

**Reply by LeXXik** [reply]
Ah, that did the trick. Thanks!
